Neuro/Tasks/TasksHandler.cs

```diff
@@ -73,6 +73,64 @@ public void UpdatePathToTask(PlayerTask task = null)
         }
     }
 
+    [HideFromIl2Cpp]
+    public PlayerTask GetClosestTask()
+    {
+        PlayerTask closestTask = null;
+        float closestDistance = Mathf.Infinity;
+
+        foreach (PlayerTask t in PlayerControl.LocalPlayer.myTasks)
+        {
+            if (!t.IsComplete && t.HasLocation)
+            {
+                Vector2[] path = NeuroPlugin.Instance.Pathfinding.FindPath(PlayerControl.LocalPlayer.transform.position, t.Locations[0]);
+                // Evaluate length of path
+                float distance = 0f;
+                for (int i = 0; i < path.Length - 1; i++)
+                {
+                    distance += Vector2.Distance(path[i], path[i + 1]);
+                }
+
+                if (distance < closestDistance)
+                {
+                    closestDistance = distance;
+                    closestTask = t;
+                }
+            }
+        }
+
+        return closestTask;
+    }
+
+    [HideFromIl2Cpp]
+    public PlayerTask GetFurthestTask()
+    {
+        PlayerTask furthestTask = null;
+        float furthestDistance = 0f;
+
+        foreach (PlayerTask t in PlayerControl.LocalPlayer.myTasks)
+        {
+            if (!t.IsComplete && t.HasLocation)
+            {
+                Vector2[] path = NeuroPlugin.Instance.Pathfinding.FindPath(PlayerControl.LocalPlayer.transform.position, t.Locations[0]);
```

Comment:
Use `.At()` extension method
```suggestion
                Vector2[] path = NeuroPlugin.Instance.Pathfinding.FindPath(PlayerControl.LocalPlayer.transform.position, t.Locations.At(0));
```